src/jit.rs

```diff
@@ -847,6 +846,12 @@ fn emit_set_exception_kind<E: UserDefinedError>(jit: &mut JitCompiler, err: Ebpf
     emit_ins(jit, X86Instruction::store_immediate(OperandSize::S64, R10, X86IndirectAccess::Offset((std::mem::size_of::<u64>() * jit.err_kind_offset) as i32), err_kind as i64));
 }
 
+fn emit_result_is_err<E: UserDefinedError>(jit: &mut JitCompiler, result_reg: u8) {
```

Comment:
`emit_ins(jit, X86Instruction::load(OperandSize::S64, RBP, R11, X86IndirectAccess::Offset(slot_on_environment_stack(jit, EnvironmentStackSlot::OptRetValPtr))));`

at both call sites could be inlined.

Reply:
I did it this way because in theory emit_result_is_err can be used with any result, but not something I feel strongly about

Reply:
The most likely use case is in combination with `emit_rust_call()` because that is where `Result` types come from.